cypress/e2e/inventory/holdings/save-empty-statistical-code-on-holdings-page.cy.js

```diff
@@ -0,0 +1,136 @@
+import { including } from 'bigtest';
+import { randomFourDigitNumber } from '../../../support/utils/stringTools';
+import { DevTeams, Permissions, TestTypes } from '../../../support/dictionary';
+import InventoryInstances from '../../../support/fragments/inventory/inventoryInstances';
+import Users from '../../../support/fragments/users/users';
+import InventorySearchAndFilter from '../../../support/fragments/inventory/inventorySearchAndFilter';
+import InventoryInstance from '../../../support/fragments/inventory/inventoryInstance';
+import HoldingsRecordEdit from '../../../support/fragments/inventory/holdingsRecordEdit';
+import TopMenu from '../../../support/fragments/topMenu';
+import HoldingsRecordView from '../../../support/fragments/inventory/holdingsRecordView';
+import { LOCATION_NAMES } from '../../../support/constants';
+
+const testData = {
+  user: {},
+  item: {
+    instanceName: `Inventory Instance ${randomFourDigitNumber()}`,
+    itemBarcode: randomFourDigitNumber(),
+  },
+  statisticalCode: 'Book, print (books)',
+  secondStatisticalCode: 'Books, electronic (ebooks)',
+  errorMessage: 'Please select to continue',
+  calloutMessage: 'has been successfully saved.',
+};
+
+describe('Holdings', () => {
+  before('Create test data', () => {
+    cy.createTempUser([Permissions.inventoryAll.gui]).then((userProperties) => {
+      testData.user = userProperties;
+      cy.getAdminToken()
+        .then(() => {
+          cy.getInstanceTypes({ limit: 1 });
+          cy.getInstanceIdentifierTypes({ limit: 1 });
+        })
+        .then(() => {
+          cy.createInstance({
+            instance: {
+              instanceTypeId: Cypress.env('instanceTypes')[0].id,
+              title: testData.item.instanceName,
+            },
+          }).then((specialInstanceId) => {
+            testData.item.instanceId = specialInstanceId;
+          });
+        });
+
+      cy.login(testData.user.username, testData.user.password, {
+        path: TopMenu.inventoryPath,
+        waiter: InventoryInstances.waitContentLoading,
+      });
+    });
+  });
+
+  after('Delete test data', () => {
+    cy.getHoldings({
+      limit: 1,
+      query: `"instanceId"="${testData.item.instanceId}"`,
+    })
+      .then((holdings) => {
+        testData.item.holdingUUID = holdings[0].id;
+      })
+      .then(() => {
+        cy.deleteHoldingRecordViaApi(testData.item.holdingUUID);
+      });
+    InventoryInstance.deleteInstanceViaApi(testData.item.instanceId);
```

Comment:
And here also please use:
`InventoryInstances.deleteInstanceViaApi({ instance: folioInstances[0], servicePoint: testData.servicePoint, shouldCheckIn: true, });`

It also deletes holding record inside of it

Reply:
This method failed when Inctance doesn't have an item